cloud-controller-manager/do/metadata.go

```diff
@@ -17,27 +17,63 @@ limitations under the License.
 package do
 
 import (
+	"context"
 	"fmt"
 	"io/ioutil"
 	"net/http"
 	"os"
 
+	"github.com/digitalocean/godo"
+	"github.com/pkg/errors"
 	"k8s.io/klog/v2"
 )
 
-const dropletRegionMetadataURL = "http://169.254.169.254/metadata/v1/region"
+const (
+	dropletRegionMetadataURL = "http://169.254.169.254/metadata/v1/region"
+	resultsPerPage           = 50
+)
 
 // dropletRegion returns the region of the currently running program.
-func dropletRegion() (string, error) {
-	// FAKE_REGION can be specified for local testing.
-	if region := os.Getenv("FAKE_REGION"); region != "" {
-		klog.Warningf("Using fake region %q from environment variable FAKE_REGION -- this should only be set for testing purposes!", region)
-		return region, nil
+func dropletRegion(regionsService godo.RegionsService) (string, error) {
+	region := os.Getenv("REGION")
+	if region == "" {
+		return httpGet(dropletRegionMetadataURL)
+	}
+
+	validRegion, err := isValidRegion(region, regionsService)
+	if err != nil {
+		return "", err
+	}
+	if !validRegion {
+		return "", errors.New("invalid region specified")
+	}
+
+	klog.Warningf("Using region %q from environment variable", region)
+	return region, nil
+}
+
+// isValidRegion checks whether the given region is a valid DO region
+func isValidRegion(region string, regionsService godo.RegionsService) (bool, error) {
+	listOptions := &godo.ListOptions{
+		Page:    1,
+		PerPage: resultsPerPage,
+	}
+
+	regions, _, err := regionsService.List(context.Background(), listOptions)
```

Comment:
I suggest we set a timeout-based context on each `List()` invocation to ensure the request doesn't hang indefinitely in case of weird networking problems. I think 30 seconds should be enough.